### PR TITLE
Template update nf-core tools 2.13.1

### PR DESCRIPTION
**Pull request by ggabernet**
- Template update for nf-core/tools version 2.13.1
- add citation
- fix samplesheet template update

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/airrflow/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/airrflow _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -13,42 +13,92 @@ nextflow.enable.dsl = 2
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    VALIDATE & PRINT PARAMETER SUMMARY
+    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { validateParameters; paramsHelp } from 'plugin/nf-validation'
+include { AIRRFLOW  } from './workflows/airrflow'
```
Comment (mapo9) [inline main.nf:20]:
```suggestion
include { AIRRFLOW                } from './workflows/airrflow'
```

File: workflows/airrflow.nf [hunk 2]
```diff
@@ -75,252 +54,237 @@ include { FASTQ_INPUT_CHECK             } from '../subworkflows/local/fastq_inpu
 // MODULE: Installed directly from nf-core/modules
 //
 include { MULTIQC                     } from '../modules/nf-core/multiqc/main'
-include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoftwareversions/main'
+include { paramsSummaryMap       } from 'plugin/nf-validation'
+include { paramsSummaryMultiqc   } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { softwareVersionsToYAML } from '../subworkflows/nf-core/utils_nfcore_pipeline'
+include { methodsDescriptionText } from '../subworkflows/local/utils_nfcore_airrflow_pipeline'
```
Comment (mapo9) [inline workflows/airrflow.nf:60]: shall we align here?

**Review by mapo9 — APPROVED** [review]
LGTM! 🚀 
Only some alignment of the imports could be done

File: main.nf [hunk 3]
```diff
@@ -13,42 +13,92 @@ nextflow.enable.dsl = 2
...
+include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_airrflow_pipeline'
```
Comment (mashehu) [inline main.nf:21]:
```suggestion
include { AIRRFLOW                } from './workflows/airrflow'
include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_airrflow_pipeline'
```

File: subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+//
+def toolCitationText() {
+    // TODO nf-core: Optionally add in-text citation tools to this list.
+    // Can use ternary operators to dynamically construct based conditions, e.g. params["run_xyz"] ? "Tool (Foo et al. 2023)" : "",
+    // Uncomment function in methodsDescriptionText to render in MultiQC report
+    def citation_text = [
+            "Tools used in the workflow included:",
+            "FastQC (Andrews 2010),",
+            "MultiQC (Ewels et al. 2016)",
+            "."
```
Comment (mashehu) [inline subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf:190]: feel free to add more tools here

File: subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        ].join(' ').trim()
+
+    return citation_text
+}
+
+def toolBibliographyText() {
+    // TODO nf-core: Optionally add bibliographic entries to this list.
+    // Can use ternary operators to dynamically construct based conditions, e.g. params["run_xyz"] ? "<li>Author (2023) Pub name, Journal, DOI</li>" : "",
```
Comment (mashehu) [inline subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf:198]: same here

File: subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+    meta["doi_text"] = meta.manifest_map.doi ? "(doi: <a href=\'https://doi.org/${meta.manifest_map.doi}\'>${meta.manifest_map.doi}</a>)" : ""
+    meta["nodoi_text"] = meta.manifest_map.doi ? "": "<li>If available, make sure to update the text to include the Zenodo DOI of version of the pipeline used. </li>"
+
+    // Tool references
+    meta["tool_citations"] = ""
+    meta["tool_bibliography"] = ""
+
+    // TODO nf-core: Only uncomment below if logic in toolCitationText/toolBibliographyText has been filled!
+    // meta["tool_citations"] = toolCitationText().replaceAll(", \\.", ".").replaceAll("\\. \\.", ".").replaceAll(", \\.", ".")
+    // meta["tool_bibliography"] = toolBibliographyText()
```
Comment (mashehu) [inline subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf:224]: then do this 🙂

File: workflows/airrflow.nf (hunk 2 above)
Comment (mashehu) [reply workflows/airrflow.nf:60]: 🍬

File: workflows/airrflow.nf [hunk 7]
```diff
@@ -75,252 +54,237 @@ include { FASTQ_INPUT_CHECK             } from '../subworkflows/local/fastq_inpu
...
+
+                ch_presto_filterseq_logs        = SEQUENCE_ASSEMBLY.out.presto_filterseq_logs
+                ch_presto_maskprimers_logs      = SEQUENCE_ASSEMBLY.out.presto_maskprimers_logs
+                ch_presto_pairseq_logs          = SEQUENCE_ASSEMBLY.out.presto_pairseq_logs
+                ch_presto_clustersets_logs      = SEQUENCE_ASSEMBLY.out.presto_clustersets_logs
+                ch_presto_buildconsensus_logs   = SEQUENCE_ASSEMBLY.out.presto_buildconsensus_logs
+                ch_presto_postconsensus_pairseq_logs = SEQUENCE_ASSEMBLY.out.presto_postconsensus_pairseq_logs
+                ch_presto_assemblepairs_logs    = SEQUENCE_ASSEMBLY.out.presto_assemblepairs_logs
+                ch_presto_collapseseq_logs      = SEQUENCE_ASSEMBLY.out.presto_collapseseq_logs
+                ch_presto_splitseq_logs         = SEQUENCE_ASSEMBLY.out.presto_splitseq_logs
```
Comment (mashehu) [inline workflows/airrflow.nf:131]: https://nf-co.re/docs/contributing/modules#what-is-the-harshil-alignment

**Review by mashehu — APPROVED** [review]

File: subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf (hunk 6 above)
Comment (ggabernet) [reply subworkflows/local/utils_nfcore_airrflow_pipeline/main.nf:224]: thank you, I hadn't noticed this! Updated now :)

**Pull request merged**